### ISSUE TITLE
"I started the timer in time but derailed anyway!"

### ISSUE THREAD

**Issue by dreeves** [beeminder/beedroid-issues#155]
Say it's 5 minutes till midnight (or whatever deadline) and I have 2 minutes of breakdancing to do for my beemergency. I start the timer and then 2 minutes in, I'm good. It's still 3 minutes till my deadline and I could hit submit now and dispatch the beemergency.

But I'm on a roll and keep breakdancing! 🕺 For 6 whole minutes! I don't notice that it's now 1 minute after my deadline, but it shouldn't matter. I breakdanced above and beyond. I hit submit at 12:01am and, sad trombone. 🐼  I've derailed because I'm submitting this 6 minutes of breakdancing for what's now officially the next day, even though almost all of it actually happened on the previous/correct day.

What to actually do without violating [anti-magic](https://blog.beeminder.com/magic)?

1. **Status quo**: Tough cookies, the goal derailed when the deadline hit and you hadn't submitted your data. The timer is just a tool to figure out what to submit -- you still have to actually submit it on time.
2. **Full magic**: If your deadline hits when a timer is running, Beedroid submits the current amount for you just under the wire. Hitting submit after that just updates that datapoint with the final amount recorded.
3. **Simply generous**: Beedroid submits the datapoint with the date corresponding to when the timer was started. You'll still probably derail at the deadline but you can maybe eke by if you submit some seconds or minutes after the deadline but before the derailment has a chance to get processed. (The length of that window depends a lot on how popular a timezone you're in.)
4. **Cry wolf**: Nudge the user to submit on time.

So... "simply generous" only matters to the extent that eking in datapoints works. So that's too gross. And "full magic" sounds hard. How does Beedroid know exactly when to submit your partial datapoint? And then it has to update the existing datapoint? What if you abort the timer and don't submit it? Does it delete the partial datapoint and retroactively derail you? Oy. As for "cry wolf", you can tell by how I named it that I don't have high hopes for it. The live countdown to derailment is already right there at the top of the page. What else is it supposed to do? Flash and beep?

I guess that leaves "status quo"! Ok, good talk.

User-me is slightly grumpy (yes, this scenario played out for me, minus the breakdancing) but, y'know, users, amirite?

*This gissue can be for pointing to the next time this happens to someone and they mistakenly think there's something Beedroid should've done about it!*

**Comment by krugerk** [comment]
I agree - "works as expected". Make it easy, make it binary, and default to one or the other. 

My take on this too as a user is - my target was to have the two minutes of activity as a datapoint by the deadline and not to have started an activity that includes also two minutes of said activity before the deadline. 

This is something an individual user can experiment with. My suggestion would be either initially or after the first time this happens, the user should update the fine print to define for the user the user's expectations of what should be accepted (by the user).

As we see in the example, the user already waited until the last minute (or last five minutes). Five minutes to go? 50-60% planned for the activity itself (the 2 minute dancing and 30 second data entry), another 20% for the unplanned or unexpected, and the remaining 20% for spontaneity. The leaves max 3 minutes for the activity (which is fortunate because we see here it probably takes a minimum of 2.5minutes anyway),  1 minute for anything unplanned, and the other minute for creativity. Use your growth mindset. Learn from it. Better luck next time.
